python/foglamp/services/south/server.py

```diff
@@ -233,8 +234,23 @@ def run(self):
         """
         _LOGGER.info('Configuration has changed for South plugin {}'.format(self._name))
 
-        # plugin shutdown before re-configure
-        self._plugin.plugin_shutdown(self._plugin_handle)
+        # Shutdown plugin and Ingest before re-configure
+        try:
+            self._plugin.plugin_shutdown(self._plugin_handle)
+        except Exception:
+            _LOGGER.exception("Unable to stop plugin '{}' during reconfigure".format(self._name))
+            raise web.HTTPInternalServerError(reason="reconfigure error - unable to stop plugin {}".format(self._name))
+        try:
+            await Ingest.stop()
+            _LOGGER.info('Stopped the Ingest server.')
+        except Exception as ex:
+            _LOGGER.exception('Unable to stop the Ingest server. %s', str(ex))
+            raise web.HTTPInternalServerError(reason="reconfigure error - unable to stop ingest for plugin {}".format(self._name))
+
+        # Cancel all pending asyncio tasks after a timeout occurs
+        done, pending = await asyncio.wait(asyncio.Task.all_tasks(), timeout=5)
```

Comment:
better could have been, plugin collect its own tasks and cancel pending on it shutdown. **It looks like, here we have just simulated the south service stop start (except its aiohttp instance), not the child plugin of it**